src/main/java/uk/gov/hmcts/reform/iacaseapi/infrastructure/clients/ClientResponseErrorHandler.java

```diff
@@ -0,0 +1,51 @@
+package uk.gov.hmcts.reform.iacaseapi.infrastructure.clients;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URI;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.client.ClientHttpResponse;
+import org.springframework.web.client.DefaultResponseErrorHandler;
+
+@Slf4j
+public class ClientResponseErrorHandler extends DefaultResponseErrorHandler  {
```

Comment:
if you could add `@Component` annotation then you have a way to inject dependency by constructor in `CcdCaseAssignment` class